ntbs-service/Pages/Migration/Index.cshtml.cs

```diff
@@ -33,10 +48,49 @@ public IActionResult OnGetAsync()
 
         public async Task<IActionResult> OnPostAsync()
         {
+            if (!ModelState.IsValid)
+            {
+                return Page();
+            }
+
             var requestId = HttpContext.TraceIdentifier;
 
-            Results = await _service.ImportNotificationsAsync(requestId, NotificationId);
+            if (UploadedFile != null)
+            {
+                var notificationIds = await GetIdListFromFile(UploadedFile);
+                var IdGroups = splitList(notificationIds, 10);
+
+                foreach (var idGroup in IdGroups)

```

Comment:
In this instance is this a `GroupId` or a `NotificationId`? Either way I think it could be named more towards this.

Reply:
Perehaps we should be using the word `batch` here?
But also as we discussed keeping parallisation at 1 seems fine given the amount of record we care about, at which point perhaps we only want to have one batch here rather than many?